src/Integration/Binding/BindingWorkflow.cs

```diff
@@ -170,26 +182,55 @@ private ProgressStepDefinition[] CreateWorkflowSteps(IProgressController control
             }
         }
 
-        internal /*for testing purposes*/ void SilentSaveSolutionIfDirty()
+        internal /*for testing purposes*/ void DownloadBindingParameters(IProgressController controller, CancellationToken token, IProgressStepExecutionEvents notifications)
         {
-            bool saved = VsShellUtils.SaveSolution(this.host, silent: true);
-            Debug.Assert(saved, "Should not be cancellable");
+            notifications.ProgressChanged(Strings.PreparingBindingWorkflowProgessMessage, double.NaN);
+
+            var properties = this.host.SonarQubeService.GetProperties(token);
+
+            if (token.IsCancellationRequested)
+            {
+                AbortWorkflow(controller, token);
+                return;
+            }
+
+            var testProjRegexProperty = properties.FirstOrDefault(x => StringComparer.Ordinal.Equals(x.Key, ServerProperty.TestProjectRegexKey));
+            var testProjRegexPattern = testProjRegexProperty?.Value ?? ServerProperty.TestProjectRegexDefaultValue;
```

Comment:
The server will return a default value if the user hasn't specified a value so I don't think this can happen in practice.
Also, the default value is determined by the C# plugin, which could change it at any point (it's already been changed once in response to user feedback). This information shouldn't be duplicated in the client-side code. It would be better to treat null/empty as meaning "no exclusions".